src/main/java/net/mcreator/element/types/interfaces/IOtherModElementsDependent.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.types.interfaces;
+
+import net.mcreator.element.parts.procedure.Procedure;
+import net.mcreator.generator.mapping.MappableElement;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public interface IOtherModElementsDependent {
+
+	default Collection<String> getUsedElementNames() {

```

Comment:
You should specify what is the string format for this. 

It is also unclear to me why both getUsedElementNames and getUsedElementMappings exist, where both methods can return vanilla or custom ME elements

Reply:
>It is also unclear to me why both getUsedElementNames and getUsedElementMappings exist, where both methods can return vanilla or custom ME elements

`getUsedElementNames()` is for things like GUIs or dimensions where only name matters for our mappings while `getUsedElementMappings()` is for rather more complex entries like `MItemBlock` or `EntityEntry`.

Reply:
If nothing else, it should be stated in JavaDOC what case is used when, and that getUsedElementNames should return values not already provided by either getUsedElementMappings or getUsedProcedures and what format String needs

Reply:
On my way to figure all that out.